### PR TITLE
Fix chain rule for log_scale gradient in projection backward pass

### LINKED ISSUE #432: Missing chain rule for log_scales gradient in projection backward pass
## Summary

The backward pass for Gaussian projection computes `dL/d(scale)` but outputs it directly as `dL/d(log_scale)` without applying the chain rule multiplication by `scale`. This causes scale gradients to be ~5× larger than they should be, leading to different training dynamics compared to GSplat.

## Root Cause

fVDB's `GaussianProjectionBackward.cu` takes `log_scales` as input, computes `scale = exp(log_scales)` inside the CUDA kernel, and then computes `dL/d(scale)`. However, it outputs this directly to `outDLossDScales` without multiplying by `scale` to get the correct `dL/d(log_scale)`.

For `scale = exp(log_scale)`, the chain rule requires:
```
dL/d(log_scale) = dL/d(scale) × d(scale)/d(log_scale) = dL/d(scale) × scale
```

## Comparison with GSplat

**GSplat architecture:**
1. Stores `log_scales` in Python, converts to `scales = torch.exp(log_scales)` before calling CUDA
2. CUDA kernel computes gradient w.r.t. `scales`
3. PyTorch autograd automatically applies chain rule: `v_log_scales = v_scales * scales`

**fVDB architecture:**
1. Passes `log_scales` directly to CUDA kernel
2. CUDA kernel computes `scale = exp(log_scales)` internally
3. CUDA kernel computes gradient w.r.t. `scale`
4. **BUG:** Outputs `dL/d(scale)` as if it were `dL/d(log_scale)` without multiplication

### fVDB Forward (GaussianProjectionBackward.cu lines 175-177):
```cpp
scale = Vec3<T>(exp(logScales[0]), exp(logScales[1]), exp(logScales[2]));
```

### fVDB Backward (GaussianUtils.cuh lines 195-198):
```cpp
// Computes gradient w.r.t. SCALE (same math as GSplat)
const nanovdb::math::Vec3<T> dLossDScale(
    R[0][0] * dLossDM[0][0] + R[1][0] * dLossDM[1][0] + R[2][0] * dLossDM[2][0],
    R[0][1] * dLossDM[0][1] + R[1][1] * dLossDM[1][1] + R[2][1] * dLossDM[2][1],
    R[0][2] * dLossDM[0][2] + R[1][2] * dLossDM[1][2] + R[2][2] * dLossDM[2][2]);
```

### fVDB Output (GaussianProjectionBackward.cu lines 271-273):
```cpp
// BUG: Writes dL/dScale as if it were dL/dLogScale
atomicAdd_system(outDLossDScales, dLossDScale[0]);      // Should be: dLossDScale[0] * scale[0]
atomicAdd_system(outDLossDScales + 1, dLossDScale[1]);  // Should be: dLossDScale[1] * scale[1]
atomicAdd_system(outDLossDScales + 2, dLossDScale[2]);  // Should be: dLossDScale[2] * scale[2]
```

### GSplat Reference (Utils.cuh lines 255-260):
```cpp
// GSplat computes the same gradient w.r.t. SCALE
v_scale[0] += R[0][0] * v_M[0][0] + R[0][1] * v_M[0][1] + R[0][2] * v_M[0][2];
v_scale[1] += R[1][0] * v_M[1][0] + R[1][1] * v_M[1][1] + R[1][2] * v_M[1][2];
v_scale[2] += R[2][0] * v_M[2][0] + R[2][1] * v_M[2][1] + R[2][2] * v_M[2][2];
// But GSplat returns v_scale, and PyTorch autograd handles the chain rule
```

## Evidence

A comparison script (see attached `rasterizer_comparison.py`) renders identical Gaussians with both rasterizers and compares gradients:

```
=== Backward Pass Comparison ===
Parameter            fVDB Norm       GSplat Norm     Ratio      Cosine Sim   Mean Diff
------------------------------------------------------------------------------------
means                0.015086        0.015086        1.0000     1.000000     0.000000
log_scales           0.024570        0.004958        4.9552     0.808861     0.000027
quats                0.000000        0.000000        0.9146     0.040777     0.000000
logit_opacities      0.002145        0.002145        1.0000     1.000000     0.000000
sh0                  0.003100        0.003100        1.0000     1.000000     0.000000
shN                  0.012006        0.012007        1.0000     1.000000     0.000000
```

Key findings:
- **Forward pass**: Identical (color diff < 0.00001)
- **Most gradients match**: means, opacities, SH coefficients have cosine_sim = 1.0
- **Scale gradients differ by ~5×**: fVDB's log_scale gradient is 4.95× larger with cosine_sim = 0.81

The ~5× ratio matches the expected error: with typical scale values around 0.2, fVDB gradients are larger by factor of `1/scale ≈ 5`.

## Impact

This bug causes:
- Scale parameters to update ~5× more aggressively than intended
- Different optimization dynamics compared to GSplat
- Potentially slower convergence in early training

## Proposed Fix

In `GaussianUtils.cuh`, modify `quaternionAndScaleToCovarianceVectorJacobianProduct` to multiply by scale:

```cpp
// Current (incorrect):
const nanovdb::math::Vec3<T> dLossDScale(...);
return {dLossDQuat, dLossDScale};

// Fixed (correct gradient for log_scale):
const nanovdb::math::Vec3<T> dLossDLogScale(
    scale[0] * (R[0][0] * dLossDM[0][0] + R[1][0] * dLossDM[1][0] + R[2][0] * dLossDM[2][0]),
    scale[1] * (R[0][1] * dLossDM[0][1] + R[1][1] * dLossDM[1][1] + R[2][1] * dLossDM[2][1]),
    scale[2] * (R[0][2] * dLossDM[0][2] + R[1][2] * dLossDM[1][2] + R[2][2] * dLossDM[2][2]));
return {dLossDQuat, dLossDLogScale};
```

Note: The same fix is needed in `GaussianProjectionJaggedBackward.cu` which has the same issue.

### PR DESCRIPTION
**Pull request by harrism**
### Summary

Fixes #432

The `quaternionAndScaleToCovarianceVectorJacobianProduct` function was computing gradients with respect to scale but the backward pass outputs gradients with respect to log-scale. Since $s = \exp(\ell)$ where $s$ is scale and $\ell$ is log-scale, the chain rule requires:

$$\frac{\partial L}{\partial \ell} = \frac{\partial L}{\partial s} \cdot \frac{\partial s}{\partial \ell} = \frac{\partial L}{\partial s} \cdot s$$

The fix multiplies each scale gradient component by the corresponding scale value.

### Changes

**`GaussianUtils.cuh`:**
- Updated docstring to clarify the function computes gradients for log-scale (not scale)
- Fixed gradient computation to apply chain rule: multiply by scale

**`GaussianProjectionBackward.cu` / `GaussianProjectionJaggedBackward.cu`:**
- Renamed variable `dLossDScale` → `dLossDLogScale` for clarity

**`GaussianProjectionBackwardTest.cpp`:**
- Removed outdated debug output (wrapped in `#if 0`)
- Updated tolerance comments to reflect new gradient ranges after the fix

### Verification

Compared fVDB and GSplat rasterizers on identical Gaussian inputs:

| Parameter | Cosine Similarity (Before) | Cosine Similarity (After) |
|-----------|---------------------------|--------------------------|
| means | 1.000 | 1.000 |
| log_scales | **0.016** | **1.000** |
| quats | 1.000 | 1.000 |
| logit_opacities | 1.000 | 1.000 |

Before the fix, the log-scale gradients had only 1.6% correlation with GSplat (essentially random). After the fix, they are identical.

### Test plan

The tests need this test data PR to be merged: https://github.com/voxel-foundation/fvdb-test-data/pull/17

- [x] `GaussianProjectionBackwardTest` passes
- [x] Gradient comparison with GSplat shows identical results

### CODE REVIEW

**Review by matthewdcong — APPROVED** [review]
Good catch, thanks!

File: src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu [hunk 1]
```diff
@@ -205,23 +205,23 @@ jaggedProjectionBackwardKernel(
...
             gpuAtomicAdd(outDLossDQuats, dLossDQuat[0]);
             gpuAtomicAdd(outDLossDQuats + 1, dLossDQuat[1]);
             gpuAtomicAdd(outDLossDQuats + 2, dLossDQuat[2]);
             gpuAtomicAdd(outDLossDQuats + 3, dLossDQuat[3]);
-            gpuAtomicAdd(outDLossDScales, dLossDScale[0]);
-            gpuAtomicAdd(outDLossDScales + 1, dLossDScale[1]);
-            gpuAtomicAdd(outDLossDScales + 2, dLossDScale[2]);
+            gpuAtomicAdd(outDLossDScales, dLossDLogScale[0]);
+            gpuAtomicAdd(outDLossDScales + 1, dLossDLogScale[1]);
+            gpuAtomicAdd(outDLossDScales + 2, dLossDLogScale[2]);
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu]: Fixed

File: src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu [hunk 2]
```diff
@@ -205,23 +205,23 @@ jaggedProjectionBackwardKernel(
             gpuAtomicAdd(outDLossDCovars + 5, dLossDCovar[2][2]);
         }
     } else {
-        // Directly output gradients w.r.t. the quaternion and scale
+        // Directly output gradients w.r.t. the quaternion and log_scale
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu]: Fixed

File: src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu (hunk 1 above)
[the remark this answers is not in the document]
Comment (harrism) [reply src/fvdb/detail/ops/gsplat/GaussianProjectionJaggedBackward.cu]: Great catch. Fixed.

File: src/tests/GaussianProjectionBackwardTest.cpp [hunk 3]
```diff
@@ -471,14 +470,15 @@ TEST_F(GaussianProjectionBackwardTestFixture, TestOrthographicProjection) {
...
     double atol = tol2.second;
     EXPECT_TRUE(torch::allclose(dLossDMeans, expectedDLossDMeans, rtol, atol));
     // EXPECT_TRUE(torch::allclose(dLossDCovars, expectedDLossDCovars, rtol, atol));
     // We use fairly large atol here since the original data was generated using scales not
     // log(scales) so there is some numerical drift.
     // The range of dLossDQuats is around [-20, 20] so atol=0.005 is reasonable
-    // The range of dLossDScales is around [-1700, 800] so atol=0.05 is very reasonable
+    // The range of dLossDScales is around [-2, 0.1] so atol=0.05 is very reasonable
     EXPECT_TRUE(torch::allclose(dLossDQuats, expectedDLossDQuats, rtol, 0.005));
     EXPECT_TRUE(torch::allclose(dLossDScales, expectedDLossDScales, rtol, 0.05));
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/tests/GaussianProjectionBackwardTest.cpp]: Fixed.
